backend/services/event_service.py

```
@@ -0,0 +1,71 @@
+        return local_dt_with_tz.astimezone(ZoneInfo("UTC"))  # Convert to UTC
+    except Exception as e:
+        raise HTTPException(status_code=400, detail=f"Invalid date/time format: {str(e)}")
+
+
+def try_convert_datetime(date: str, time: str, tz: str, fallback: datetime) -> datetime:
+    if date and time and tz:
+        return convert_to_utc(date, time, tz)
+    return fallback
+
+# Helper function to update post and event fields
+def update_fields(fields, model_instance, db: Session) -> bool:
```

Ruff (0.8.2)

Missing type annotation for function argument `fields`

(ANN001)